packages/transactions/src/new-compile-transaction.ts

```diff
@@ -4,12 +4,30 @@ import { SignatureBytes } from '@solana/keys';
 import {
     CompilableTransactionMessage,
     getCompiledTransactionMessageEncoder,
+    IDurableNonceTransactionMessage,
+    ITransactionMessageWithBlockhashLifetime,
     newCompileTransactionMessage,
 } from '@solana/transaction-messages';
 
+import { TransactionBlockhashLifetime, TransactionDurableNonceLifetime, TransactionWithLifetime } from './lifetime';
 import { NewTransaction, OrderedMap, TransactionMessageBytes } from './transaction';
 
-export function compileTransaction(transactionMessage: CompilableTransactionMessage): NewTransaction {
+export function compileTransaction(
+    transactionMessage: ITransactionMessageWithBlockhashLifetime &
+        Omit<CompilableTransactionMessage, 'lifetimeConstraint'>,
+): NewTransaction & { lifetimeConstraint: TransactionBlockhashLifetime };
```

Comment:
Heh, this confused me for a minute, but I think I see what you're doing here. You do the specific lifetime-type then omit `lifetimeConstraint` from `CompilableTransactionMessage` to ensure it's the specific lifetime, right?

If so, might it be a bit cleaner to just do this?

```ts
export function compileTransaction(
    transactionMessage: CompilableTransactionMessage & { lifetimeConstraint: TransactionBlockhashLifetime }
): NewTransaction & { lifetimeConstraint: TransactionBlockhashLifetime };
```

Reply:
Not sure why I thought I needed the `Omit`, removed it, thanks!